### PR TITLE
React query implementation

### PR DESCRIPTION
**Pull request by himanshu-spr**

### CODE REVIEW

File: src/hooks/useRepository.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    filteredData = {
+      id: data.id,
+      name: data.name,
+      full_name: data.full_name,
+      avatar: data.owner.avatar_url,
+      url: data.html_url,
+      stars: data.stargazers_count,
+      description: data.description,
+    };
+  }
```
Comment (NiketMalik) [inline src/hooks/useRepository.ts]: wrap in `useMemo`

File: src/hooks/useRepositories.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        id: repository.id,
+        name: repository.name,
+        full_name: repository.full_name,
+        avatar: repository.owner.avatar_url,
+        url: repository.html_url,
+        stars: repository.stargazers_count,
+        description: repository.description,
+      };
+    });
+  }
```
Comment (NiketMalik) [inline src/hooks/useRepositories.ts]: wrap in useMemo

File: src/helpers/index.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Repository } from "../interfaces";
+
+export function isError(error: unknown): error is Error {
+  return error instanceof Error;
+}
```
Comment (NiketMalik) [inline src/helpers/index.ts]: Why is this required?

File: src/components/Favorites/FavoriteList/FavoriteListItem/index.tsx [hunk 4]
```diff
@@ -3,30 +3,47 @@ import { FavoriteListItemProps } from "../../../../interfaces";
...
-        <td className="fav-description-data">{favorite.description}</td>
-        <td className="fav-stars-data">{favorite.stars}</td>
-        <td>
-          <button className="remove-button" onClick={removeHandler}>
-            Remove
-          </button>
-        </td>
-      </tr>
-    </>
+    data && (
```
Comment (NiketMalik) [inline src/components/Favorites/FavoriteList/FavoriteListItem/index.tsx]: Better approach is:

```suggestion
    if(!data) {
      return null;
    }
    
    return <>....</>
```

File: src/components/HomePage/RepositoryList/SortDropdown/index.tsx [hunk 5]
```diff
@@ -1,77 +1,30 @@
...
-} from "../../../../redux";
-
 import { SORT_TYPES } from "../../../../constants/sort";
+import { setSortValue } from "../../../../redux";
 
 const getOptions = () => {
   const options = Object.values(SORT_TYPES);
-  return options.map((option) => <option value={option}>{option}</option>);
+  return options.map((option) => (
+    <option key={option} value={option}>
```
Comment (NiketMalik) [inline src/components/HomePage/RepositoryList/SortDropdown/index.tsx]: missing key

File: src/hooks/useRepository.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { useQuery } from "react-query";
+import axios from "axios";
+import { FavoriteData, RepoResponse, Repository } from "../interfaces";
+
+export default function useRepository(favorite: FavoriteData) {
+  let searchString = "https://api.github.com/repos/" + favorite.fullname;
+
+  const { status, data, error } = useQuery(
```
Comment (NiketMalik) [inline src/hooks/useRepository.ts]:
```suggestion
  const { status, data, error } = useQuery<Repository>(
```

Your data will infer type `Repository | undefined` then

**Review by NiketMalik — CHANGES_REQUESTED** [review]

File: src/components/HomePage/RepositoryList/SortDropdown/index.tsx (hunk 5 above)
Comment (himanshu-spr) [reply src/components/HomePage/RepositoryList/SortDropdown/index.tsx]: key is present

File: src/helpers/index.ts (hunk 3 above)
Comment (himanshu-spr) [reply src/helpers/index.ts]: in the component where I was using the error. It is giving type error. `Object is of type unknown`

File: src/components/Favorites/FavoriteList/FavoriteListItem/index.tsx [hunk 7]
```diff
@@ -1,51 +1,44 @@
...
 
+  if (status === "loading") {
+    return <p className="loading">Loading...</p>;
+  }
+
+  if (status === "error" && error) {
+    return <p className="error">{error.message}</p>;
+  }
+
+  if (!data) return null;
```
Comment (NiketMalik) [inline src/components/Favorites/FavoriteList/FavoriteListItem/index.tsx]: Expose `isError` and `isLoading` from react-query directly instead.

File: src/components/HomePage/index.tsx [hunk 8]
```diff
@@ -1,24 +1,24 @@
...
-  if (repositoryData.loading) {
+  if (status === "loading") {
     return <p className="loading">Loading...</p>;
   }
 
-  if (repositoryData.error) {
-    return <p className="error">{repositoryData.error}</p>;
+  if (status === "error" && error) {
+    return <p className="error">{error.message}</p>;
   }
```
Comment (NiketMalik) [inline src/components/HomePage/index.tsx]: Expose `isError` and `isLoading` from react-query directly instead.

**Review by NiketMalik — CHANGES_REQUESTED** [review]

**Review by NiketMalik — APPROVED** [review]
